[Thread-8] WEBORB ERROR:8/3/2010 10:05:22:System.NullReferenceException: Object reference not set to an instance of an object.
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at GerenteCacheBancoDadosCliente.getCacheByIdCorp(Int32 idCorp) in C:\dev\sistemadaempresa\ambiente_desenvolvimento\marcos\lado_servidor\sde_marcos\SDE\code\pattern\GerenteCacheBancoDadosCliente.cs:line 42
   at GerenteCacheBancoDadosCliente.getAtualizacoes(Int32 idCorp, Int32 ultima) in C:\dev\sistemadaempresa\ambiente_desenvolvimento\marcos\lado_servidor\sde_marcos\SDE\code\pattern\GerenteCacheBancoDadosCliente.cs:line 65
   at SDE.CamadaNuvem.NuvemNotificacoes.Lista_Notificacoes(Int32 idCorp, Int32 ultima) in C:\dev\sistemadaempresa\ambiente_desenvolvimento\marcos\lado_servidor\sde_marcos\SDE\code\CamadaNuvem\NuvemNotificacoes.cs:line 18
[Thread-4] WEBORB ERROR:8/3/2010 10:28:33:Object reference not set to an instance of an object.    at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at Weborb.Util.License.LicenseManager.GetInstance(Int32 productID)
   at Weborb.Util.Invocation.invoke(Object obj, MethodInfo method, Object[] arguments)
   at Weborb.Handler.ObjectHandler.invoke(String typeName, String function, Object[] arguments, RequestContext context)
[Thread-8] WEBORB ERROR:8/3/2010 10:29:14:Object reference not set to an instance of an object.    at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at Weborb.Util.License.LicenseManager.GetInstance(Int32 productID)
   at Weborb.Util.Invocation.invoke(Object obj, MethodInfo method, Object[] arguments)
   at Weborb.Handler.ObjectHandler.invoke(String typeName, String function, Object[] arguments, RequestContext context)
[Thread-4] WEBORB ERROR:8/3/2010 10:30:03:Object reference not set to an instance of an object.    at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at Weborb.Util.License.LicenseManager.GetInstance(Int32 productID)
   at Weborb.Util.Invocation.invoke(Object obj, MethodInfo method, Object[] arguments)
   at Weborb.Handler.ObjectHandler.invoke(String typeName, String function, Object[] arguments, RequestContext context)
[Thread-8] WEBORB ERROR:8/3/2010 10:30:54:An item with the same key has already been added.    at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at Weborb.Util.License.LicenseManager.GetInstance(Int32 productID)
   at Weborb.Util.Invocation.invoke(Object obj, MethodInfo method, Object[] arguments)
   at Weborb.Handler.ObjectHandler.invoke(String typeName, String function, Object[] arguments, RequestContext context)
[Thread-4] WEBORB ERROR:8/3/2010 11:01:04:An item with the same key has already been added.    at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at Weborb.Util.License.LicenseManager.GetInstance(Int32 productID)
   at Weborb.Util.Invocation.invoke(Object obj, MethodInfo method, Object[] arguments)
   at Weborb.Handler.ObjectHandler.invoke(String typeName, String function, Object[] arguments, RequestContext context)
[Thread-12] WEBORB ERROR:8/3/2010 11:03:55:An item with the same key has already been added.    at System.ThrowHelper.ThrowArgumentException(ExceptionResource resource)
   at System.Collections.Generic.Dictionary`2.Insert(TKey key, TValue value, Boolean add)
   at System.Collections.Generic.Dictionary`2.Add(TKey key, TValue value)
   at Weborb.Util.License.LicenseManager.GetInstance(Int32 productID)
   at Weborb.Util.Invocation.invoke(Object obj, MethodInfo method, Object[] arguments)
   at Weborb.Handler.ObjectHandler.invoke(String typeName, String function, Object[] arguments, RequestContext context)
[Thread-4] WEBORB ERROR:8/3/2010 12:13:56:Db4objects.Db4o.Ext.DatabaseFileLockedException: C:\dev\sistemadaempresa\ambiente_desenvolvimento\marcos\banco_de_dados\Banco\0.dbf ---> System.IO.IOException: The process cannot access the file 'C:\dev\sistemadaempresa\ambiente_desenvolvimento\marcos\banco_de_dados\Banco\0.dbf' because it is being used by another process.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at Sharpen.IO.RandomAccessFile..ctor(String file, Boolean readOnly, Boolean lockFile)
   at Db4objects.Db4o.IO.RandomAccessFileFactory.NewRandomAccessFile(String path, Boolean readOnly, Boolean lockFile)
   --- End of inner exception stack trace ---
   at Db4objects.Db4o.IO.RandomAccessFileFactory.NewRandomAccessFile(String path, Boolean readOnly, Boolean lockFile)
   at Db4objects.Db4o.IO.FileStorage.FileBin..ctor(BinConfiguration config)
   at Db4objects.Db4o.IO.FileStorage.Open(BinConfiguration config)
   at Db4objects.Db4o.IO.StorageDecorator.Open(BinConfiguration config)
   at Db4objects.Db4o.IO.CachingStorage.Open(BinConfiguration config)
   at Db4objects.Db4o.Internal.IoAdaptedObjectContainer.OpenImpl()
   at Db4objects.Db4o.Internal.ObjectContainerBase._IRunnable_104.Run()
   at Db4objects.Db4o.Foundation.DynamicVariable.With(Object value, IRunnable block)
   at Db4objects.Db4o.Foundation.Environments.RunWith(IEnvironment environment, IRunnable runnable)
   at Db4objects.Db4o.Internal.ObjectContainerBase.WithEnvironment(IRunnable runnable)
   at Db4objects.Db4o.Internal.ObjectContainerBase.Open()
   at Db4objects.Db4o.Internal.IoAdaptedObjectContainer..ctor(IConfiguration config, String fileName)
   at Db4objects.Db4o.Internal.ObjectContainerFactory.OpenObjectContainer(IConfiguration config, String databaseFileName)
   at Db4objects.Db4o.Db4oFactory.OpenFile(IConfiguration config, String databaseFileName)
   at Db4objects.Db4o.Db4oFactory.OpenFile(String databaseFileName)
   at GerenteConectividadeBancoDados.get(Int32 idCorp, TipoBanco tipoBanco) in C:\dev\sistemadaempresa\ambiente_desenvolvimento\marcos\lado_servidor\sde_marcos\SDE\code\pattern\GerenteConectividadeBancoDados.cs:line 99
   at SDE.CamadaNuvem.SuperNuvem..ctor() in C:\dev\sistemadaempresa\ambiente_desenvolvimento\marcos\lado_servidor\sde_marcos\SDE\code\CamadaNuvem\SuperNuvem.cs:line 17
   at SDE.CamadaNuvem.NuvemNotificacoes..ctor()
[Thread-9] WEBORB ERROR:8/3/2010 12:13:56:Db4objects.Db4o.Ext.DatabaseFileLockedException: C:\dev\sistemadaempresa\ambiente_desenvolvimento\marcos\banco_de_dados\Banco\0.dbf ---> System.IO.IOException: The process cannot access the file 'C:\dev\sistemadaempresa\ambiente_desenvolvimento\marcos\banco_de_dados\Banco\0.dbf' because it is being used by another process.
   at System.IO.__Error.WinIOError(Int32 errorCode, String maybeFullPath)
   at System.IO.FileStream.Init(String path, FileMode mode, FileAccess access, Int32 rights, Boolean useRights, FileShare share, Int32 bufferSize, FileOptions options, SECURITY_ATTRIBUTES secAttrs, String msgPath, Boolean bFromProxy)
   at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share)
   at Sharpen.IO.RandomAccessFile..ctor(String file, Boolean readOnly, Boolean lockFile)
   at Db4objects.Db4o.IO.RandomAccessFileFactory.NewRandomAccessFile(String path, Boolean readOnly, Boolean lockFile)
   --- End of inner exception stack trace ---
   at Db4objects.Db4o.IO.RandomAccessFileFactory.NewRandomAccessFile(String path, Boolean readOnly, Boolean lockFile)
   at Db4objects.Db4o.IO.FileStorage.FileBin..ctor(BinConfiguration config)
   at Db4objects.Db4o.IO.FileStorage.Open(BinConfiguration config)
   at Db4objects.Db4o.IO.StorageDecorator.Open(BinConfiguration config)
   at Db4objects.Db4o.IO.CachingStorage.Open(BinConfiguration config)
   at Db4objects.Db4o.Internal.IoAdaptedObjectContainer.OpenImpl()
   at Db4objects.Db4o.Internal.ObjectContainerBase._IRunnable_104.Run()
   at Db4objects.Db4o.Foundation.DynamicVariable.With(Object value, IRunnable block)
   at Db4objects.Db4o.Foundation.Environments.RunWith(IEnvironment environment, IRunnable runnable)
   at Db4objects.Db4o.Internal.ObjectContainerBase.WithEnvironment(IRunnable runnable)
   at Db4objects.Db4o.Internal.ObjectContainerBase.Open()
   at Db4objects.Db4o.Internal.IoAdaptedObjectContainer..ctor(IConfiguration config, String fileName)
   at Db4objects.Db4o.Internal.ObjectContainerFactory.OpenObjectContainer(IConfiguration config, String databaseFileName)
   at Db4objects.Db4o.Db4oFactory.OpenFile(IConfiguration config, String databaseFileName)
   at Db4objects.Db4o.Db4oFactory.OpenFile(String databaseFileName)
   at GerenteConectividadeBancoDados.get(Int32 idCorp, TipoBanco tipoBanco) in C:\dev\sistemadaempresa\ambiente_desenvolvimento\marcos\lado_servidor\sde_marcos\SDE\code\pattern\GerenteConectividadeBancoDados.cs:line 99
   at SDE.CamadaNuvem.SuperNuvem..ctor() in C:\dev\sistemadaempresa\ambiente_desenvolvimento\marcos\lado_servidor\sde_marcos\SDE\code\CamadaNuvem\SuperNuvem.cs:line 17
   at SDE.CamadaNuvem.NuvemNotificacoes..ctor()
